src/mame/sinclair/spec_snqk.cpp

```diff
@@ -1766,6 +1775,358 @@ void spectrum_state::snx_decompress_block(address_space &space, uint8_t *source,
 	}
 }
 
+static u16 hrust_decompress_block(uint8_t *dest, uint8_t *source, uint16_t size)
+{
```

Comment:
You should at least make the `source` data `const` to avoid accidents.  Also, is the size here the available source data, or the space at the destination?  You really need both for safety.